driver-core/src/test/functional/com/mongodb/ClusterFixture.java

```diff
@@ -379,18 +377,14 @@ public static OperationContext createNewOperationContext(final TimeoutSettings t
     }
 
     private static ReadWriteBinding getBinding(final Cluster cluster,
-            final ReadPreference readPreference,
-            final OperationContext operationContext) {
+                                               final ReadPreference readPreference,
+                                               final OperationContext operationContext) {
         if (!BINDING_MAP.containsKey(readPreference)) {
-            ReadWriteBinding binding = new ClusterBinding(cluster, readPreference, ReadConcern.DEFAULT, operationContext);
-            if (serverVersionAtLeast(3, 6)) {
-                binding = new SessionBinding(binding);
-            }
+            ReadWriteBinding binding = new SessionBinding(new ClusterBinding(cluster, readPreference, ReadConcern.DEFAULT,
+                    operationContext));
             BINDING_MAP.put(readPreference, binding);
         }
-        ReadWriteBinding readWriteBinding = BINDING_MAP.get(readPreference);
-        return new OperationContextBinding(readWriteBinding,
-                operationContext.withSessionContext(readWriteBinding.getOperationContext().getSessionContext()));
+        return BINDING_MAP.get(readPreference);
```

Comment:
Is this return statement intentional?

The merge I got here (and below) instead of this return was:
```
        ReadWriteBinding readWriteBinding = BINDING_MAP.get(readPreference);
        return new OperationContextBinding(readWriteBinding,
                operationContext.withSessionContext(readWriteBinding.getOperationContext().getSessionContext()));
```

Reply:
This was not intentional. I have corrected it to use:

`      ReadWriteBinding readWriteBinding = BINDING_MAP.get(readPreference);
        return new OperationContextBinding(readWriteBinding,
                operationContext.withSessionContext(readWriteBinding.getOperationContext().getSessionContext()));`
  
  and the equivalent in the async version as well. Thanks!